fl4health/utils/dataset_converter.py

```diff
@@ -0,0 +1,159 @@
+from abc import ABC
+from typing import Callable, Optional, Tuple, Union
+
+import torch
+
+from fl4health.utils.dataset import BaseDataset
+
+
+class DatasetConverter(ABC, BaseDataset):
```

Comment:
I think I might be a little confused as to why the DatasetConverter here is inheriting from BaseDataset? I might be mistaken, but the DatasetConverter seems like it's job it to "encode" and "decode" a torch dataset into one that is compatible with the autoencoder training. So it's not a dataset per se. Let me know I'm mixing something there.

Reply:
Ah I see now. I missed that part in this class when working through it. Thanks for the explanation!